source/libspirv.cpp

```diff
@@ -111,8 +112,15 @@ bool SpirvTools::Validate(const uint32_t* binary,
 bool SpirvTools::Validate(const uint32_t* binary, const size_t binary_size,
                           const ValidatorOptions& options) const {
   spv_const_binary_t the_binary{binary, binary_size};
-  return spvValidateWithOptions(impl_->context, options, &the_binary,
-                                nullptr) == SPV_SUCCESS;
+  spv_diagnostic diagnostic = nullptr;
```

Comment:
Does spvValidateWIthOptions create the diagnostic? That's an ... odd interface

Reply:
Yes it does, which is why you pass it a pointer to a pointer.  It is odd.

Reply:
Ugh.